app/commands/put_content_with_links.rb

```diff
@@ -77,10 +77,23 @@ def content_item_attributes
     end
 
     def create_or_update_links!
-      LinkSet.create_or_replace(content_id: content_id, links: content_item[:links]) do |link_set|
+      LinkSet.create_or_replace(content_id: content_id) do |link_set|
         version = Version.find_or_initialize_by(target: link_set)
         version.increment
         version.save! if link_set.valid?
+
```

Comment:
Did you consider wrapping this whole block of code in a database transaction?

```
ActiveRecord::Base.transaction do
  link_set.links.delete_all
  # link creation logic
end
```

This way if a link creation fails for some reason your database will remain in the before the transaction.